libs/designer/src/lib/core/state/undoRedo/undoRedoTypes.ts

```diff
@@ -0,0 +1,62 @@
+import { updateNodeConnection } from '../../actions/bjsworkflow/connections';
+import type { RootState } from '../../store';
+import {
+  addEdgeFromRunAfter,
+  addImplicitForeachNode,
+  addNode,
+  addSwitchCase,
+  deleteNode,
+  deleteSwitchCase,
+  moveNode,
+  pasteNode,
+  pasteScopeNode,
+  removeEdgeFromRunAfter,
+  updateRunAfter,
+} from '../workflow/workflowSlice';
+
+export interface StateHistory {
+  past: string[];
+  future: string[];
+}
+
+// Omitted slices: designerView, designerOptions, dev, undoRedo
+export type UndoRedoPartialRootState = Pick<
```

Comment:
Looking for inputs here on whether some other slice could be removed too. I was unsure about "customCode" and "workflowParameters".

Reply:
I think custom code and workflowparameters should both stay, and I think naming here is fine